test/users.js

```diff
@@ -0,0 +1,200 @@
+const chai = require('chai'),
+  dictum = require('dictum.js'),
+  server = require('./../app'),
+  should = chai.should(),
+  errors = require('../app/errors'),
+  User = require('../app/models').users;
+
+const validateError = ({ actualError, expectedMessage, expectedInternalCode }) => {
+  actualError.should.have.status(400);
+  actualError.response.should.be.json;
+  actualError.response.body.should.have.property('message');
+  actualError.response.body.should.have.property('internalCode');
+
+  const { message, internalCode } = actualError.response.body;
+  message.should.be.a('array');
+  message.should.have.lengthOf(1);
+  message[0].should.equal(expectedMessage);
+  internalCode.should.equal(expectedInternalCode);
+};
+
+describe('users', () => {
+  describe('/users POST', () => {
+    it('should fail because email is not wolox format', done => {
+      chai
+        .request(server)
+        .post('/users')
+        .send({
+          firstName: 'firstName',
+          lastName: 'lastName',
+          password: 'password',
+          email: 'email@gmail.co'
+        })
+        .end(err => {
+          validateError({
+            actualError: err,
+            expectedMessage: 'The email has an invalid format',
+            expectedInternalCode: errors.CREATE_USER_ERROR
+          });
+          done();
+        });
+    });
+    it('should fail because email is not unique', done => {
+      const user = {
+        firstName: 'firstName',
+        lastName: 'lastName',
+        password: 'password',
+        email: 'unique@wolox.co'
+      };
+      User.createModel(user).then(userCreated => {
+        chai
+          .request(server)
+          .post('/users')
+          .send(user)
+          .end(err => {
+            validateError({
+              actualError: err,
+              expectedMessage: 'The email must be unique',
+              expectedInternalCode: errors.CREATE_USER_ERROR
+            });
+            done();
+          });
+      });
+    });
+    it('should fail because password is less than 8 characters', done => {
+      chai
+        .request(server)
+        .post('/users')
+        .send({
+          firstName: 'firstName',
+          lastName: 'lastName',
+          password: 'short',
+          email: 'email@wolox.co'
+        })
+        .end(err => {
+          validateError({
+            actualError: err,
+            expectedMessage: 'The password is invalid. It must be alphanumeric and a minimum of 8 characters',
```

Comment:
You are hardcoding this message several times, better define it as a constant outside the tests